prometheus/src/main/scala/filodb/prometheus/parse/Parser.scala

```diff
@@ -31,8 +31,7 @@ object Parser extends StrictLogging {
     }
   }
 
-  // Only called directly by tests.
-  def parseQuery(query: String): Expression = {
+  private def parseQueryWithoutMetricNameRequirement(query: String): Expression = {
```

Comment:
The validations "setup" by each pre-PR `Parser` method (i.e. that exactly one metric name exists for any selector) remain the same as before. The only difference is that the validations are guaranteed to run before the `Parser` returns a result.

Only `parseQueryWithoutMetricNameRequirement`-- a private method-- can return a parsed result where a metric name has not been enforced. Its only current use is within `queryToLabelMap` below.